### PR TITLE
Add initial remote form

### PR DESCRIPTION
**Pull request by nixocio**
Add initial remote form, modify related files. This one will be expanded to deal with edge cases, and yaml/json upload field.

See: https://issues.redhat.com/browse/AAH-2493

image

### CODE REVIEW

**Review by AlexSCorey — CHANGES_REQUESTED** [review]
This looks great.  I have just proposed a few changes.

File: frontend/hub/remotes/RemoteForm.tsx [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+      />
+      <PageForm<IRemotes>
+        submitText={t('Create remote')}
+        onSubmit={onSubmit}
+        onCancel={() => navigate(-1)}
+        defaultValue={{ name: '', url: '' }}
+      >
+        <>
+          <RemoteInputs />
+          <PageFormSection singleColumn>
```
[the remark this answers is not in the document]
Comment (nixocio) [reply frontend/hub/remotes/RemoteForm.tsx:43]: @AlexSCorey, that is a good suggestion, let me take a look.

**Review by AlexSCorey — APPROVED** [review]
Looks good to me.

**Review by vidyanambiar — COMMENTED** [review]
Nice work 🎉  

Curious, why does the URL require a slash at the end? I was able to save a URL without slash in ansible-hub-ui.
- If the URL does not end in `/` an error is displayed, but this error does not go away when the slash is added at the end of the URL
image

- Can we add checks to validate URL? Currently it is possible to just save `/` as the value of URL.

File: frontend/hub/remotes/RemoteForm.tsx [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+        type="password"
+        name="proxy_password"
+        label={t('Proxy password')}
+        placeholder={t('Enter a proxy password')}
+      />
+    </>
+  );
+}
+
+function CeriticatesAdvancedRemoteInputs() {
```
[the remark this answers is not in the document]
Comment (nixocio) [reply frontend/hub/remotes/RemoteForm.tsx:86]: @vidyanambiar the ansible-hub-ui frontend client is appending a `/` to the URL. I will add the same logic. The trailing slash is required by design on the backend.

**Review by vidyanambiar — APPROVED** [review]
As discussed offline URL validation will be added in follow on PRs. This looks good to me. 👍

**Pull request merged**